api/api.proto

```diff
@@ -1203,6 +1203,83 @@ service OpenStorageRole {
       };
     }
 }
+
+// OpenStorageFilesystemTrimService provides methods to manage a
+// filesystem trim operation on a volume. This operation runs in the
+// background on a mounted volume. If the volumes are not mounted, these API 
+// return error.
+//
+service OpenStorageFilesystemTrimService {
+
+  // Start a filesystem Trim background operation on a mounted volume
+  rpc Start(SdkFilesystemTrimStartRequest)
+    returns (SdkFilesystemTrimStartResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-trim/start"
+        body: "*"
+      };
+    }
+
+  // Get Status of a filesystem Trim background operation on a mounted volume, if any
+  rpc GetStatus(SdkFilesystemTrimGetStatusRequest)
+    returns (SdkFilesystemTrimGetStatusResponse){
+      option(google.api.http) = {
+        get: "/v1/filesystem-trim/getstatus/query/{volume_id}/{mount_path}"
```

Comment:
This should probably be `get: "/v1/filesystem-trim/getstatus"`. The caller would then send `"GET" /v1/filesystem-trim/getstatus?volume_id=myid&mount_path=mypath`